mlflow/gateway/providers/bedrock.py

```diff
@@ -0,0 +1,279 @@
+import json
+import os
+import time
+from enum import Enum
+
+import boto3
+import botocore.config
+import botocore.exceptions
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.config import AWSBedrockConfig, AWSIdAndKey, AWSRole, RouteConfig
+from mlflow.gateway.constants import (
+    MLFLOW_AI_GATEWAY_ANTHROPIC_DEFAULT_MAX_TOKENS,
+    MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS,
+)
+from mlflow.gateway.exceptions import AIGatewayConfigException
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import rename_payload_keys, send_request
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class AWSBedrockModelProvider(Enum):
+    AMAZON = "amazon"
+    COHERE = "cohere"
+    AI21 = "ai21"
+    ANTHROPIC = "anthropic"
+
+    @classmethod
+    def of_str(cls, name: str):
+        name = name.lower()
+
+        for opt in cls:
+            if opt.name.lower() == name or opt.value.lower() == name:
+                return opt
+
+
+class AWSBedrockProvider(BaseProvider):
+    def __init__(self, config: RouteConfig):
+        super().__init__(config)
+
+        if config.model.config is None or not isinstance(config.model.config, AWSBedrockConfig):
+            raise TypeError(f"Invalid config type {config.model.config}")
+        self.bedrock_config: AWSBedrockConfig = config.model.config
+        self._client = None
+        self._client_created = 0
+
+    def _client_expired(self):
+        if not isinstance(self.bedrock_config.aws_config, AWSRole):
+            return False
+
+        return (
+            (time.monotonic_ns() - self._client_created)
+            >= (self.bedrock_config.aws_config.session_length_seconds) * 1_000_000_000,
+        )
+
+    def get_bedrock_client(self):
+        if self._client is not None and not self._client_expired():
+            return self._client
+
+        session_args, config_args, client_args = {}, {}, {}
+        aws_config = self.bedrock_config.aws_config
+
+        region_name = aws_config.aws_region or os.environ.get(
```

Comment:
Could these environment variables be migrated to `environment_variables.py` (https://github.com/mlflow/mlflow/blob/master/mlflow/environment_variables.py) and have a distinct naming convention that identifies what they are used for `MLFLOW_AI_GATEWAY_BEDROCK_AWS_REGION` `MLFLOW_AI_GATEWAY_BEDROCK_AWS_DEFAULT_REGION` ?

Reply:
Actually, on second thought - why are these able to be overridden in the Provider definition? Is this intended? Similar logic for handling env var defined api keys for other providers is handled within the route constructor validation within `config.py` so that the route definitions remain immutable. 
Is there a distinct need to be mutate the route handling within the Gateway Server after a route has been defined in order to change the AWS Region?

Reply:
The goal here is for defer as much as possible to the normal boto credential provider chain for the benefit of the AWSRole path, while still allowing a RouteConfig to provide credentials.  Given that, I think we can just remove all the explicit AWS_* environment variable stuff and count on boto to Do The Right Thing.

Reply:
Sounds great!